src/datasource.js

```diff
@@ -25,16 +26,26 @@ export class HawkularDatasource {
       "counter": "counters",
       "availability": "availability"
     };
-    let variables = new Variables(templateSrv);
-    this.capabilitiesPromise = this.queryVersion()
-      .then(version => new Capabilities(version));
-    this.queryProcessor = new QueryProcessor($q, backendSrv, variables, this.capabilitiesPromise, this.url, this.headers, this.typeResources);
+    this.variables = new Variables(templateSrv);
+    this.capabilitiesPromise = this.queryVersion().then(version => new Capabilities(version));
+    this.queryProcessor = new QueryProcessor($q, backendSrv, this.variables, this.capabilitiesPromise, this.url, this.headers, this.typeResources);
   }
 
   query(options) {
     let validTargets = options.targets
       .filter(target => !target.hide)
-      .filter(target => (target.queryBy === 'tags' && target.tags.length > 0) || target.target !== 'select metric');
+      .map(target => {
+        if (target.id === undefined && target.target !== 'select metric') {
+          // backward compatibility
+          target.id = target.target;
+        } else if (target.id === '-- none --') {
+          delete target.id;
+        }
+        return target;
+      })
+      .filter(target => target.id !== undefined
+         || (target.tags !== undefined && target.tags.length > 0)
+         || (target.tagsQL !== undefined && target.tagsQL.length > 0));
```

Comment:
This is something I fear (probably due to ignorance) when I write javascript: if I write `if (target.id)` instead of `if (target.id !== undefined)`, what happens if `target.id` is 0? or the string "0" ?